src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,200 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> previous;
+        private Node<T> next;
+
+        public Node(T value, Node<T> previous, Node<T> next) {
+            this.value = value;
+            this.previous = previous;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size < 1) {
+            head = new Node<>(value, null, null);
+            tail = head;
+        } else {
+            Node<T> nodeToAdd = new Node<>(value, tail, null);
+            tail.next = nodeToAdd;
+            tail = nodeToAdd;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIfIndexSuitableForAdding(index);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> nodeOnGivenIndex = getNode(index);
+            if (isHead(nodeOnGivenIndex)) {
+                Node<T> nodeToAdd = new Node<>(value, null, nodeOnGivenIndex);
+                nodeOnGivenIndex.previous = nodeToAdd;
+                head = nodeToAdd;
+            } else {
+                Node<T> previousNode = nodeOnGivenIndex.previous;
+                Node<T> nodeToAdd = new Node<>(value, previousNode, nodeOnGivenIndex);
+                previousNode.next = nodeToAdd;
+                nodeOnGivenIndex.previous = nodeToAdd;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNode(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> nodeToRemove = getNode(index);
+        unlink(nodeToRemove);
+        return nodeToRemove.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> current = head;
+
+        boolean removed = false;
+        do {
+            if (valuesAreEqual(object, current)) {
+                unlink(current);
+                removed = true;
+                break;
+            }
+            current = current.next;
+        } while (current != tail);
+
+        return removed;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size < 1;
```

Comment:
Minor, but from my perspective, it would be more natural to do
`return size == 0;`
as our size definitely would not go under zero